src/main.cpp

```diff
@@ -232,189 +245,161 @@ void benchmark_all_multi_taint(antlr4::ANTLRInputStream& prog, std::string file_
-            time_func("Analyzing: ", 
-                cpu_analysis::worklist, nodes, analyis_info.node_to_index, analyis_info.transfers);
-            cpu_watch.save_time<Microseconds>();
-            std::vector<StatefulNode<std::set<std::string>>> cpu_nodes = create_states<std::set<std::string>>(program.nodes, {TAINT_VAR});
-            set_bit_vector_state(nodes, cpu_nodes);
-
-            init_gpu();
-
-            std::cout << "\n⭐ bit-cuda analysis ⭐" << std::endl;
-            program = parse_to_cfg_transformer(prog, call_counts);
-            Stopwatch bit_cuda_watch;
-            auto bit_cuda_nodes = bit_cuda_analysis(program);
-            bit_cuda_watch.save_time<Microseconds>();
-
-            if(!state_equality(cpu_nodes, bit_cuda_nodes)){
-                std::cout << "###### cpu != bit-cuda #####" << std::endl;
-            }
-
-            std::cout << "\n⭐ bit-cuda worklist analysis ⭐" << std::endl;
-            program = parse_to_cfg_transformer(prog, call_counts);
-            Stopwatch cuda_worklist_watch;
-            auto cuda_worklist_nodes = bit_cuda_worklist_analysis(program);
-            cuda_worklist_watch.save_time<Microseconds>();
+        std::stringstream header;
+        if (call_counts.find("f") == call_counts.end()) {
+            header << file_name;
+        }
+        else {
+            header << file_name << " : " << call_counts["f"];
+        }
+        auto program = parse_to_cfg_transformer(prog, call_counts);
+        std::cout << "CFG Nodes: " << program.nodes.size() << "\n" << std::endl;
+        header << ";" << program.nodes.size();
+        Stopwatch::add_header(header.str());
+
+        std::cout << "\n⭐ CPU analysis ⭐" << std::endl;
+        reduce_variables(program.entryNodes);
+        Stopwatch cpu_watch;
+        TransferCreator analyis_info = get_analysis_information(program.nodes);
+        std::vector<StatefulNode<cpu_analysis::BitVector>> nodes = create_states<cpu_analysis::BitVector>(program.nodes, cpu_analysis::BitVector(1));
+        time_func("Analyzing: ", 
+            cpu_analysis::worklist, nodes, analyis_info.node_to_index, analyis_info.transfers);
+        cpu_watch.save_time<Microseconds>();
+        std::vector<StatefulNode<std::set<std::string>>> cpu_nodes = create_states<std::set<std::string>>(program.nodes, {TAINT_VAR});
+        set_bit_vector_state(nodes, cpu_nodes);
+
+        init_gpu();
+
+        std::cout << "\n⭐ bit-cuda analysis ⭐" << std::endl;
+        program = parse_to_cfg_transformer(prog, call_counts);
+        Stopwatch bit_cuda_watch;
+        auto bit_cuda_nodes = bit_cuda_analysis(program);
+        bit_cuda_watch.save_time<Microseconds>();
+
+        if(!state_equality(cpu_nodes, bit_cuda_nodes)){
+            std::cout << "###### cpu != bit-cuda #####" << std::endl;
+        }
 
-            if(!state_equality(bit_cuda_nodes, cuda_worklist_nodes)){
```

Comment:
we should still compare bit-cuda and cuda worklist

Reply:
Comparing the two is actually impossible currently. We dont have any easy way to get the proper information compare the two results. This will have to wait until cpu solutions also get updated